src/wp_filter_sender.cpp

```diff
@@ -0,0 +1,244 @@
+#include <mavros_msgs/CommandTOL.h>
+#include <mavros_msgs/State.h>
+#include <nav_msgs/Odometry.h>
+#include <geometry_msgs/Pose.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <cmath>
+#include <math.h>
+#include <ros/ros.h>
+#include <std_msgs/Float64.h>
+#include <std_msgs/String.h>
+#include <mavros_msgs/CommandBool.h>
+#include <mavros_msgs/SetMode.h>
+#include <mavros_msgs/PositionTarget.h>
+#include <unistd.h>
+#include <vector>
+#include <ros/duration.h>
+#include <iostream>
+
+#include <task_master/Waypoint.h>
+#include <task_master/TaskGoalPosition.h>
+
+#include "waypointSenderFunctionsLib.h"
+
+/**
+\defgroup control_functions
+This module is designed to make high level control programming more simple. 
+*/
+mavros_msgs::State current_state;
+nav_msgs::Odometry current_pose;
+geometry_msgs::Pose correction_vector;
+geometry_msgs::Point local_offset_pose;
+geometry_msgs::PoseStamped desired_waypoint;
+
+float current_heading;
+float local_offset;
+float correction_heading = 0;
+float local_desired_heading; 
+
+waypoint goal_pos;
+
+
+ros::Publisher local_pos_pub;
+ros::Subscriber currentPos;
+ros::Subscriber state_sub;
+ros::ServiceClient arming_client;
+ros::ServiceClient set_mode_client;
+ros::Subscriber goal_pos_sub;
+
+//get armed state
+void state_cb(const mavros_msgs::State::ConstPtr& msg)
+ {
+   current_state = *msg;
+ }
+
+// set goal pose 
+void goal_pose(const task_master::TaskGoalPosition& msg){
+    goal_pos.x = msg.x;
+    goal_pos.y = msg.y;
+    goal_pos.z = msg.z;
+    goal_pos.psi = 0.0;
+ }
+
+float pose_cb(const nav_msgs::Odometry::ConstPtr& msg)
+{
+  current_pose = *msg;
+  enu_2_local(current_pose);
+  float q0 = current_pose.pose.pose.orientation.w;
+  float q1 = current_pose.pose.pose.orientation.x;
+  float q2 = current_pose.pose.pose.orientation.y;
+  float q3 = current_pose.pose.pose.orientation.z;
+  float psi = atan2((2*(q0*q3 + q1*q2)), (1 - 2*(pow(q2,2) + pow(q3,2))) );
+  float current_heading = psi*(180/M_PI);
+  return current_heading;
+
+}
+
+float get_current_heading()
+ {
+ 	return current_heading;
+ }
+
+
+/*
+@returns 0 - connected to fcu 
+@returns -1 - failed to connect to drone
+*/
+int wait4connect()
+{
+	ROS_INFO("Waiting for FCU connection");
+	// wait for FCU connection
+	while (ros::ok() && !current_state.connected)
+	{
+		ros::spinOnce();
+		ros::Duration(0.01).sleep();
+	}
+	if(current_state.connected)
+	{
+		ROS_INFO("Connected to FCU");	
+		return 0;
+	}else{
+		ROS_INFO("Error connecting to drone");
+		return -1;	
+	}
+	
+	
+}
+
+/**
+@returns 0 - mission started
+@returns -1 - failed to start mission
+*/
+int wait4start()
```

Comment:
can we change the name to waitForGuided? more descriptive of what is actually going on.